src/main/scala/edu/berkeley/ce/rockslicing/Block.scala

```diff
@@ -199,23 +214,42 @@ case class Block(center: (Double,Double,Double), faces: Seq[Face]) {
         val newY = NumericUtils.roundToTolerance(centerY + y)
         val newZ = NumericUtils.roundToTolerance(centerZ + z)
         val updatedFaces = updateFaces(newX, newY, newZ)
-        if (translatedJoint.d < 0.0) {
-          Vector(
-            // New origin is guaranteed to lie within joint, so initial d = 0
-            Block((newX,newY,newZ), Face((-translatedJoint.a, -translatedJoint.b, -translatedJoint.c), 0.0,
-                  translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces),
-            Block((newX,newY,newZ), Face((translatedJoint.a,translatedJoint.b,translatedJoint.c), 0.0, 
-                  translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces)
-          )
+
+        // New origin is guaranteed to lie within joint, so initial d = 0 for all child blocks
+        val childBlockA = if (translatedJoint.d < 0.0) {
+          Block((newX,newY,newZ), Face((-translatedJoint.a, -translatedJoint.b, -translatedJoint.c), 0.0,
+            translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces)
+        } else {
+          Block((newX,newY,newZ), Face((translatedJoint.a, translatedJoint.b, translatedJoint.c), 0.0,
+            translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces)
+        }
+        val childBlockB = if (translatedJoint.d < 0.0) {
+          Block((newX,newY,newZ), Face((translatedJoint.a,translatedJoint.b,translatedJoint.c), 0.0,
+            translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces)
         } else {
-          Vector(
-            // New origin is guaranteed to lie within joint, so initial d = 0
-            Block((newX,newY,newZ), Face((translatedJoint.a, translatedJoint.b, translatedJoint.c), 0.0,
-                  translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces),
-            Block((newX,newY,newZ), Face((-translatedJoint.a,-translatedJoint.b,-translatedJoint.c), 0.0,
-                  translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces)
-          )
+          Block((newX,newY,newZ), Face((-translatedJoint.a,-translatedJoint.b,-translatedJoint.c), 0.0,
+            translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces)
         }
+
+        var childBlocks = Vector(childBlockA, childBlockB)
```

Comment:
Yeah, my instinct is always to avoid using `var` if I can. The problem is that there would be four different branches to take at this point in the code if I were to use conditionals and `return`.

My take is that the code is more concise and more clear using a `var` here, and the `var` is only being used for a few lines of code.